torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1519,12 +1519,15 @@ void InitXlaModuleBindings(py::module m) {
                                  const py::list& tile_assignment,
                                  bool replicated = false, bool manual = false) {
     TORCH_LAZY_COUNTER("XlaMarkSharding", 1);
+    XLATensorPtr xtensor = bridge::GetXlaTensor(input);
+    XLAGraphExecutor::Get()->UnregisterTensor(xtensor->data().get());
```

Comment:
I think this unregister approach will conflict with the Clone pr @steventk-g is working on.

Reply:
So what will happen if we don't register, will compiler returns us 2 duplicate output?

Reply:
The node hashing work is to avoid hash collision at the node level; this helps us from confusing two computation graphs, sharded and unsharded, with the same hash.

This is to sanitize sync tensors and the parameters lists. The registration happens at the device level, and we register one tensor to one XLATensor. If we don't register.. then I think we are still associating the unsharded XLATensor to the input tensor, which is not correct. And we can't or should be careful trying to associate two XLATensors to a tensor... let's talk more, I feel like there is much cost to our goal.

Reply:
Synced offline, we will register, but not unregister to rely on the reference counter.